### PR TITLE
[FIX] project: add strict comparison for date_deadline in decoration-danger

### PR DESCRIPTION
**Pull request by rajp-odoo**
Steps to reproduce:
- install project module.
- open form view of any task
- create new task and set value of deadline
- deadline is getting text-decoration even if deadline is not passed.

Sol:
- Condition ensures date_deadline is not None and strictly less than current_date
- This change improves the robustness of the condition by handling undefined date_deadline values.

task: 3970136

### CODE REVIEW

**Review by deneuvillem — COMMENTED** [review]
LGTM :raised_hands: Thanks for your work!

**Review by xavierbol — COMMENTED** [review]
Thanks for your work! 🙂 

I left some minor comments

File: addons/project/views/project_task_views.xml [hunk 1]
```diff
@@ -353,7 +353,8 @@
                             <field name="partner_id" nolabel="0" widget="res_partner_many2one" class="o_task_customer_field" invisible="not project_id"/>
                             <label for="date_deadline"/>
                             <div id="date_deadline_and_recurring_task" class="d-inline-flex w-100">
-                                <field name="date_deadline" nolabel="1" decoration-danger="date_deadline &lt; current_date and state not in ['1_done', '1_canceled']"/>
+                                <field name="date_deadline" nolabel="1"
+                                       decoration-danger="date_deadline and (date_deadline &lt; current_date) and state not in ['1_done', '1_canceled']"/>
```
[the remark this answers is not in the document]
Comment (rajp-odoo) [reply addons/project/views/project_task_views.xml]: Yeah, I have reduced the diff now and yes parenthesis is not required so I have removed.

**Comment by rajp-odoo** [comment]
hello @xavierbol 
I have reduced the diff now,

Thanks for the review

**Review by xavierbol — COMMENTED** [review]
LGTM, thanks again for your work!

robodoo r+

**Pull request closed without merging**